index.js

```diff
@@ -36,18 +34,39 @@ export default async function getStream(inputStream, options = {}) {
 			}
 		}
 
-		return getBufferedValue();
+		return getBufferedValue(chunks, isBuffer, length);
 	} catch (error) {
-		try {
-			error.bufferedData = getBufferedValue();
-			// This throws when the buffered data is larger than the maximum length
-			// for a string or buffer
-		} catch {}
-
+		error.bufferedData = getSafeBufferedValue(chunks, isBuffer, length);
 		throw error;
 	}
 }
 
+const getSafeBufferedValue = (chunks, isBuffer, length) => {
+	try {
+		return getBufferedValue(chunks, isBuffer, length);
+	} catch {
+		return truncateBufferedValue(chunks, isBuffer);
+	}
+};
+
+// If the input is larger than the maximum length for a string or a buffer,
+// it will fail. We retry it with increasingly smaller inputs, so that
+// `error.bufferedData` is still set, albeit with a truncated value, since that
+// is still useful for debugging.
+const truncateBufferedValue = (chunks, isBuffer) => {
+	let chunksCount = chunks.length;
+	do {
+		chunksCount = Math.floor(chunksCount / SPLIT_FACTOR);
+		try {
+			return getBufferedValue(chunks.slice(0, chunksCount), isBuffer);
+		} catch {}
+	} while (chunksCount > 0);
```

Comment:
`chunksCount > 0` should never be `false`, since when `chunksCount = 0` is called, `chunks.slice(0, 0)` will be an empty string/buffer, which is guaranteed not to throw.
So, one could use `while (true)` instead. I have used `chunksCount > 0` instead to avoid some linting warning, and as an additional safety net to avoid infinite loops, just in case.